src/Host/Client/Impl/Blobs/RBlobStream.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.R.Host.Client {
+    public class RBlobStream : Stream {
+        private readonly IRBlobInfo _blob;
+        private readonly IRBlobService _blobService;
+        private bool _isDisposed;
+
+        public override bool CanRead { get; } = true;
+        public override bool CanSeek { get; } = false;
+
+        private bool _canWrite;
+        public override bool CanWrite {
+            get {
+                return _canWrite;
+            }
+        }
+
+        private long _length;
+        public override long Length {
+            get {
+                return _length;
+            }
+        }
+
+        private RBlobStream(IRBlobInfo blob, bool canWrite, IRBlobService blobService) {
+            _blob = blob;
+            _canWrite = canWrite;
+            _blobService = blobService;
+            _isDisposed = false;
+        }
+
+        private long _position;
+        public override long Position {
+            get {
+                return _position;
+            }
+            set {
+                Seek(value, SeekOrigin.Begin);
```

Comment:
This is intentional. No one should be setting Position (except in this class) as this stream does not support it.